xmodule/tests/test_capa_block.py

```diff
@@ -1263,10 +1402,15 @@ def test_codejail_error_upon_problem_creation(self):
     def _rescore_problem_error_helper(self, exception_class):
         """Helper to allow testing all errors that rescoring might return."""
         # Create the block
-        block = CapaFactory.create(attempts=1, done=True)
+        block = CapaFactory.create(attempts=0)
+        CapaFactory.answer_key()
+
+        # Check the problem
+        get_request_dict = {CapaFactory.input_key(): '1'}
+        block.submit_problem(get_request_dict)
```

Comment:
Because we have to make at least one attempt so that the fields `self.student_answers_history` and `self.correct_map_history` are not empty [🡥](https://github.com/openedx/edx-platform/blob/862b942363576e03973f9a21e201c3626847c91a/xmodule/capa_block.py#L2301-L2302). If they are empty, the `get_grade_from_answers` method will never be executed, and therefore the exception is not thrown.